crates/astria-core/src/primitive/v1/asset/denom.rs

```diff
@@ -75,6 +75,24 @@ impl Denom {
         };
         trace
     }
+
+    /// Calculates the length of the display formatted [Denom] without allocating a String.
+    #[must_use]
+    pub fn display_len(&self) -> Option<usize> {
+        match self {
+            Denom::TracePrefixed(trace) => {
```

Comment:
Define these on the trace-prefixed and ibc-prefixed types. Then delegate to them in this enum.

Reply:
I did so with `TracePrefixed`, but clippy threw a (IMO pretty reasonable) warning about having an unused self when applying to IBC prefixed, so I kept the fixed value in `Denom::display_len()`. Lmk if this works for you!